python/llm/example/GPU/HuggingFace/Multimodal/qwen2-audio/generate.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+from io import BytesIO
+from urllib.request import urlopen
+import librosa
+import torch
+from transformers import Qwen2AudioForConditionalGeneration, AutoProcessor
+from ipex_llm import optimize_model
+
+def main(args):
+    model_path = args.repo_id_or_model_path
+
+    processor = AutoProcessor.from_pretrained(model_path)
+    model = Qwen2AudioForConditionalGeneration.from_pretrained(model_path)
+    model = optimize_model(model, low_bit='sym_int4', optimize_llm=True)
+    model = model.half()
```

Comment:
model = model.half().xpu()

Reply:
this leads to 
```bash
(qiyue-llm-0815) arda@arda-arc16:~/qiyue/my/ipex-llm/python/llm/example/GPU/HuggingFace/Multimodal/qwen2-audio$ python generate.py --repo-id-or-model-path  /mnt/disk1/models/Qwen2-Audio-7B-Instruct/
/home/arda/miniforge3/envs/qiyue-llm-0815/lib/python3.11/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: ''If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
  warn(
/home/arda/miniforge3/envs/qiyue-llm-0815/lib/python3.11/site-packages/transformers/deepspeed.py:24: FutureWarning: transformers.deepspeed module is deprecated and will be removed in a future version. Please import deepspeed modules directly from transformers.integrations
  warnings.warn(
2024-08-23 00:30:05,587 - INFO - intel_extension_for_pytorch auto imported
Loading checkpoint shards: 100%|██████████████████████████████████████████████████████████████████| 5/5 [00:03<00:00,  1.31it/s]
2024-08-23 00:30:09,869 - INFO - Converting the current model to sym_int4 format......
It is strongly recommended to pass the `sampling_rate` argument to this function. Failing to do so can result in silent errors that might be hard to debug.
We detected that you are passing `past_key_values` as a tuple and this is deprecated and will be removed in v4.43. Please use an appropriate `Cache` class (https://huggingface.co/docs/transformers/v4.41.3/en/internal/generation_utils#transformers.Cache)
Traceback (most recent call last):
  File "/home/arda/qiyue/my/ipex-llm/python/llm/example/GPU/HuggingFace/Multimodal/qwen2-audio/generate.py", line 74, in <module>
    main(args)
  File "/home/arda/qiyue/my/ipex-llm/python/llm/example/GPU/HuggingFace/Multimodal/qwen2-audio/generate.py", line 57, in main
    generate_ids = model.generate(**inputs, max_length=max_length)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/arda/miniforge3/envs/qiyue-llm-0815/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/arda/miniforge3/envs/qiyue-llm-0815/lib/python3.11/site-packages/ipex_llm/transformers/pipeline_parallel.py", line 281, in generate
    return original_generate(self,
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/arda/miniforge3/envs/qiyue-llm-0815/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/arda/miniforge3/envs/qiyue-llm-0815/lib/python3.11/site-packages/transformers/generation/utils.py", line 2024, in generate
    result = self._sample(
             ^^^^^^^^^^^^^
  File "/home/arda/miniforge3/envs/qiyue-llm-0815/lib/python3.11/site-packages/transformers/generation/utils.py", line 3020, in _sample
    next_tokens = torch.multinomial(probs, num_samples=1).squeeze(1)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: probability tensor contains either `inf`, `nan` or element < 0
```

So i am using model.half().to('xpu') instead